### PR TITLE
[WIP] Changes in sklearn wrappers for LDA and LSI models

### PR DESCRIPTION
**Pull request by chinmayapancholi13**
This PR makes the following changes in the sklearn wrappers for LDA and LSI models : 
- Doesn't train model from within the `__init__` function of the wrappers (following the convention followed by the fuctions in scikit-learn)
- Uses `super` while calling the parent class' `__init__` function from `fit` function

### CODE REVIEW

File: gensim/sklearn_integration/sklearn_wrapper_gensim_ldamodel.py [hunk 1]
```diff
@@ -46,15 +46,6 @@ def __init__(
         self.gamma_threshold = gamma_threshold
         self.minimum_probability = minimum_probability
         self.random_state = random_state
```
Comment (menshikh-iv) [inline gensim/sklearn_integration/sklearn_wrapper_gensim_ldamodel.py]: Please remove corpus parameter from constructor (you pass a corpus only for `fit*` methods) in both models.

**Comment by menshikh-iv** [comment]
The same as #1405

File: gensim/test/test_sklearn_integration.py [hunk 2]
```diff
@@ -37,69 +38,48 @@
...
         for i in range(10):
             self.model.partial_fit(X=corpus)  # fit against the model again
             doc = list(corpus)[0]  # transform only the first document
-            transformed = self.model[doc]
+            transformed = self.model.transform(doc)
             transformed_approx = matutils.sparse2full(transformed, 2)  # better approximation
-        expected = [0.13, 0.87]
+        expected = [0.87, 0.0]
         passed = numpy.allclose(sorted(transformed_approx), sorted(expected), atol=1e-1)
         self.assertTrue(passed)
```
Comment (menshikh-iv) [inline gensim/test/test_sklearn_integration.py]: It's often broken in Travis, please check, what is an reason of this

File: gensim/test/test_sklearn_integration.py [hunk 3]
```diff
@@ -191,6 +181,22 @@ def testSetGetParams(self):
         for key in param_dict.keys():
             self.assertEqual(model_params[key], param_dict[key])
 
+    def testPersistence(self):
```
Comment (menshikh-iv) [inline gensim/test/test_sklearn_integration.py]: Same as LdaSeq

File: gensim/test/test_sklearn_integration.py [hunk 4]
```diff
@@ -173,9 +163,9 @@ def testPipeline(self):
         id2word = Dictionary(map(lambda x: x.split(), data.data))
         corpus = [id2word.doc2bow(i.split()) for i in data.data]
         clf = linear_model.LogisticRegression(penalty='l2', C=0.1)
-        text_lda = Pipeline((('features', model,), ('classifier', clf)))
-        text_lda.fit(corpus, data.target)
-        score = text_lda.score(corpus, data.target)
+        text_lsi = Pipeline((('features', model,), ('classifier', clf)))
+        text_lsi.fit(corpus, data.target)
+        score = text_lsi.score(corpus, data.target)
         self.assertGreater(score, 0.50)
```
Comment (menshikh-iv) [inline gensim/test/test_sklearn_integration.py]: same as LdaSeq

File: gensim/test/test_sklearn_integration.py (hunk 3 above)
Comment (chinmayapancholi13) [inline gensim/test/test_sklearn_integration.py]: Thanks. Done.

File: gensim/test/test_sklearn_integration.py (hunk 4 above)
Comment (chinmayapancholi13) [inline gensim/test/test_sklearn_integration.py]: Thanks. Done.

File: gensim/test/test_sklearn_integration.py (hunk 2 above)
Comment (chinmayapancholi13) [inline gensim/test/test_sklearn_integration.py]: I have made changes to solve this problem and this should now be resolved. :)

**Pull request merged**